inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java

```diff
@@ -337,6 +337,11 @@ private void preProcessTemplateFileTask(TaskRequest taskRequest) {
         List<StreamSourceEntity> sourceEntities = sourceMapper.selectTemplateSourceByCluster(needCopiedStatusList,
                 Lists.newArrayList(SourceType.FILE), agentClusterName);
         sourceEntities.forEach(sourceEntity -> {
+            InlongGroupEntity groupEntity = groupMapper.selectByGroupId(sourceEntity.getInlongGroupId());
+            Set<GroupStatus> noNeedAddTask = Sets.newHashSet(GroupStatus.SUSPENDED, GroupStatus.SUSPENDING);
```

Comment:
Why no put **_noNeedAddTask_** outside the loop?

Reply:
Because `inlongGroupId` may be different in `sourceEntities`.